src/main/java/randoop/main/GenTests.java

```diff
@@ -1335,4 +1359,93 @@ private Path getResourceDirectoryPath(String resourceDirectory) {
   public void incrementSequenceCompileFailureCount() {
     this.sequenceCompileFailureCount++;
   }
+
+  /**
+   * Returns names of classes that the given classes depend on. A class is considered a dependency
+   * if it is a parameter to a method/constructor. Does not return omitted or non-accessible
+   * classes. Does not return dependencies for non-accessible methods/constructors.
+   *
+   * @param classnames names of dependent classes
+   * @param accessibility accessibility predicate
+   * @return classnames of dependencies
+   */
+  public static Set<@ClassGetName String> getDependentClassnamesFromClassnames(
+      Set<@ClassGetName String> classnames, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> dependenciesClassnames = new TreeSet<>();
+
+    for (String classname : classnames) {
+      try {
+        Class<?> getDependenciesFrom = Class.forName(classname);
+        for (Method method : getDependenciesFrom.getDeclaredMethods()) {
+          addParameterTypesIfShould(method, dependenciesClassnames, accessibility);
+        }
+        for (Constructor<?> constructor : getDependenciesFrom.getConstructors()) {
+          addParameterTypesIfShould(constructor, dependenciesClassnames, accessibility);
+        }
+      } catch (ClassNotFoundException e) {
+        throw new RandoopUsageError(String.format("Cannot load class %s", classname));
+      }
+    }
+    return dependenciesClassnames;
+  }
+
+  /**
+   * Returns names of classes that methods in {@code methodlist} depend on. Does not add
+   * dependencies of methods that should be omitted. Does not add classes that are not accessible or
+   * should be omitted.
+   *
+   * @param accessibilityPredicate an accessibility predicate
+   * @param omitMethods each regex indicates methods that should not be called directly in generated
+   *     tests
+   * @return classnames of dependencies
+   */
+  public static Set<@ClassGetName String> getDependentClassnamesFromMethodList(
```

Comment:
Please make this method take `methodList` as an argument rather than depending on the value of a field.